client/controller/index.web.js

```diff
@@ -42,17 +42,33 @@ export const ProviderWrappedLayout = ( {
 	const state = store.getState();
 	const userLoggedIn = isUserLoggedIn( state );
 
+	const [ isExperimentLoaded, setIsExperimentLoaded ] = useState( null );
+
+	async function loadDomainExperiment() {
+		const experimentAssignment = await loadExperimentAssignment(
+			'calypso_sidebar_upsell_dedicated_upgrade_flow_202301'
+		);
+		const variationName = experimentAssignment?.variationName;
+		sessionStorage.setItem( 'calypso_sidebar_upsell_experiment', variationName );
+		setIsExperimentLoaded( true );
+	}
+
+	function shouldLoadDomainExperiment() {
+		return (
+			window.location.pathname.startsWith( '/domains/add' ) ||
+			window.location.pathname.startsWith( '/plans/yearly' )
+		);
+	}
+
 	useEffect( () => {
-		async function loadExperiment() {
-			const experimentAssignment = await loadExperimentAssignment(
-				'calypso_sidebar_upsell_dedicated_upgrade_flow_202301'
-			);
-			const variationName = experimentAssignment?.variationName;
-			sessionStorage.setItem( 'calypso_sidebar_upsell_experiment', variationName );
+		if ( shouldLoadDomainExperiment() ) {
+			loadDomainExperiment();
 		}
+	}, [ currentRoute ] );
 
-		loadExperiment();
-	}, [] );
+	if ( shouldLoadDomainExperiment() && ! isExperimentLoaded ) {
```

Comment:
We need this to ensure we'll only Load the subcomponents of the `domain` and `plans` pages once the experiment was loaded. To ensure [codes like this](https://github.com/Automattic/wp-calypso/pull/72932/files#diff-e9d5bbfeb0cc388a1ec2501f08f49524a97724605eca06bc1f3f626bf12cfab4R105) works as expected

Reply:
Question. Do we also want to limit this to routes with `domainAndPlanPackage` set in the query string? For example, ATM if I'm in the treatment group and I go the domain search via the standard path (/domains/add/example.wordpress.com) I see our new design.

Reply:
That's a good question @DustyReagan. It will depend if we want to restrict to free users only. Asked here: pebzTe-Fx-p2#comment-1215

Reply:
Good catch @DustyReagan, adding our Slack thread here for future reference: p1675449243063399/1675446834.367269-slack-CKZHG0QCR

Reply:
[Added it here](https://github.com/Automattic/wp-calypso/pull/72932/commits/98fbcb52c9da99e9ed9e4299d62efad93917e4c9) @DustyReagan